tests/test_routing.py

```diff
@@ -1164,6 +1162,65 @@ def test_extract_straher_streams_watersheds_d8(self):
         watershed_vector = None
         watershed_layer = None
 
+    def test_single_drain_point(self):
+        """PGP.routing: test single_drain_point pitfill."""
+        dem_array = numpy.zeros((11, 11), dtype=numpy.float32)
+        dem_array[0, 0] = -1.0
+        dem_array[1:8, 1:8] = -2.0
+        dem_array[10, 7] = -4
+        dem_array[8:11, 8:11] = 2.0
+        dem_array[9:11, 9:11] = 1.0
+        dem_array[10, 10] = -7.0
+        dem_path = os.path.join(self.workspace_dir, 'dem.tif')
+        _array_to_raster(dem_array, None, dem_path)
+
+        # outlet tuple at 0,0, just drain one edge
+        expected_array_0_0 = numpy.copy(dem_array)
+        expected_array_0_0[1:8, 1:8] = -1
+        expected_array_0_0[10, 7] = 0
+        expected_array_0_0[8:11, 8:11] = 2.0
+
+        # output tuple at 5,5, it's a massive pit so drain the edges
+        expected_array_5_5 = numpy.copy(dem_array)
+        expected_array_5_5[8:11, 8:11] = 2.0
+        expected_array_5_5[10, 7] = 0
+
+        for output_tuple, expected_array in [
+                ((0, 0), expected_array_0_0),
+                ((5, 5), expected_array_5_5),
+                ]:
+            fill_path = os.path.join(self.workspace_dir, 'filled.tif')
+            pygeoprocessing.routing.fill_pits(
+                (dem_path, 1), fill_path,
+                single_outlet_tuple=output_tuple,
+                working_dir=self.workspace_dir)
+            result_array = pygeoprocessing.raster_to_numpy_array(fill_path)
+            numpy.testing.assert_almost_equal(result_array, expected_array)
+
+    def test_detect_lowest_drain_and_sink(self):
+        """PGP>routing: test detect_lowest_sink_and_drain."""
```

Comment:
>PGP.routing

Reply:
fixed in ff64743e1148d989d559e87db6c767465d130753